app/scripts/background.js

```diff
@@ -606,33 +611,11 @@ async function loadPhishingWarningPage() {
  */
 export async function loadStateFromPersistence() {
   // migrations
-  const migrator = new Migrator({
-    migrations,
-    defaultVersion: process.env.WITH_STATE
-      ? FIXTURE_STATE_METADATA_VERSION
-      : null,
-  });
   migrator.on('error', console.warn);
 
-  if (process.env.WITH_STATE) {
-    const stateOverrides = await generateWalletState();
-    firstTimeState = { ...firstTimeState, ...stateOverrides };
-  }
-
   // read from disk
   // first from preferred, async API:
-  versionedData =
-    (await localStore.get()) || migrator.generateInitialState(firstTimeState);
-
-  // check if somehow state is empty
-  // this should never happen but new error reporting suggests that it has
-  // for a small number of users
-  // https://github.com/metamask/metamask-extension/issues/3919
-  if (versionedData && !versionedData.data) {
-    // unable to recover, clear state
-    versionedData = migrator.generateInitialState(firstTimeState);
-    sentry.captureMessage('MetaMask - Empty vault found - unable to recover');
-  }
+  versionedData = await localStore.get();
```

Comment:
Question: `versionedData` is set twice but with same value, first time `await localStore.get();`, second time `versionedData = await migrator.migrateData(versionedData);`, can you explain to me why do we need to set again after dealing with migration error (` migrator.on('error', ...` )?

Reply:
I improved variable naming to help clarify what is happening here: 381a058eea